pkg/execute/mapping.go

```diff
@@ -0,0 +1,150 @@
+	CommandStart    CommandVerb = "start"
+	CommandStop     CommandVerb = "stop"
+	CommandStatus   CommandVerb = "status"
+	CommandConfig   CommandVerb = "config"
+)
+
+// CommandExecutor defines command structure for executors
+type CommandExecutor interface {
+	Commands() map[CommandVerb]CommandFn
+	FeatureName() FeatureName
+}
+
+// CommandFn is a single command (eg. List())
+type CommandFn func(ctx context.Context, cmdCtx CommandContext) (interactive.Message, error)
+
+// CommandContext contains the context for CommandFn
+type CommandContext struct {
+	Args            []string
+	ClusterName     string
+	CommGroupName   string
+	BotName         string
+	RawCmd          string
+	User            string
+	Conversation    Conversation
+	Platform        config.CommPlatformIntegration
+	ExecutorFilter  executorFilter
+	NotifierHandler NotifierHandler
+	Mapping         *CommandMapping
+}
+
+// FeatureName defines the name and aliases for a feature
+type FeatureName struct {
+	Name    string
+	Aliases []string
+}
+
+// CommandMapping allows to register and lookup commands and dynamically build help messages
+type CommandMapping struct {
+	commands map[CommandVerb]map[string]CommandFn
+	help     map[CommandVerb][]FeatureName
+}
+
+// NewCmdsMapping registers command and help mappings
+func NewCmdsMapping(executors []CommandExecutor) (*CommandMapping, error) {
+	mappingsErrs := multierror.New()
+	cmdsMapping := make(map[CommandVerb]map[string]CommandFn)
+	helpMapping := make(map[CommandVerb][]FeatureName)
+	for _, executor := range executors {
+		cmds := executor.Commands()
+		subCmd := executor.FeatureName()
+		for verb, cmdFn := range cmds {
+			if value := cmdsMapping[verb]; value == nil {
+				cmdsMapping[verb] = make(map[string]CommandFn)
+			}
+			if value := helpMapping[verb]; value == nil {
+				helpMapping[verb] = make([]FeatureName, 0)
+			}
+			cmdsMapping[verb][subCmd.Name] = cmdFn
+			helpMapping[verb] = append(helpMapping[verb], subCmd)
+			for _, featureName := range subCmd.Aliases {
+				if _, ok := cmdsMapping[verb][featureName]; ok {
+					mappingsErrs = multierror.Append(mappingsErrs, fmt.Errorf("command collision: tried to register '%s %s', but it already exists", verb, featureName))
+				}
+				cmdsMapping[verb][featureName] = cmdFn
+			}
+		}
+	}
+	mapping := &CommandMapping{
+		commands: cmdsMapping,
+		help:     helpMapping,
+	}
+	return mapping, mappingsErrs.ErrorOrNil()
+}
+
+// FindFn looks up CommandFn by verb and feature
+func (m *CommandMapping) FindFn(verb CommandVerb, feature string) (CommandFn, bool, bool) {
+	features, ok := m.commands[verb]
+	if !ok {
+		return nil, false, false
+	}
+	fn, ok := features[feature]
+	if !ok {
+		return nil, true, false
+	}
+	return fn, true, true
+}
+
+// HelpMessageForVerb dynamically builds help message for given CommandVerb, or empty string
+func (m *CommandMapping) HelpMessageForVerb(verb CommandVerb, botName string) string {
+	cmd, ok := m.help[verb]
+	if !ok {
+		return ""
```

Comment:
here we can just return `incompleteCmdMsg` as I mentioned in the previous comment. This will simply the code and also group it in a single place. For example, right now the `HelpMessageForVerb` func is used in 2 places, so we will make sure that each time a proper help message is returned.